omicsclaw/runtime/context_layers/__init__.py

```
@@ -733,16 +542,82 @@ def __post_init__(self) -> None:
         object.__setattr__(self, "output_style", str(self.output_style or "").strip())
 
 
+_PredicateEventSink = Callable[["events.LifecycleEvent"], None]
```

Ruff (0.15.12)

[error] Undefined name `events`

(F821)